packages/core/src/StateNode.ts

```diff
@@ -357,7 +357,7 @@ class StateNode<
       }
     }
 
-    dfs(this);
+    dfs(this as StateNode);
```

Comment:
shouldn't `dfs` parameter type be adjusted instead? 

Also if you change public `any` in the other file to `unknown` then I think the same thing could happen here for StateNode.